### PR TITLE
issue/UI support for MSK

### LINKED ISSUE #2739: UI support for MSK
Changes need to be done:

**Topic Create page**
1. Move `Replication Factor` down to the `Min In Sync Replicas` line, `Cleanup policy` goes up to the  `Number of partitions` level
2. By default `Replication Factor`, `Min In Sync Replicas`, `Maximum message size in bytes`, `Time to retain data (in ms)`  fields should not be filled. "Default" text should be rendered instead. If user didn't set any value to these fields, corresponding parameters should not be passed to backend.

**Edit setting page**
1. Topic properties (both explicit  fields and "Custom parameters") should only be rendered if `source = DYNAMIC_TOPIC_CONFIG`. This means that this parameter was explicitly set when topic was created and it should be shown. Explicit params fields should follow the same logic described in ( Topic create page. p2 ).

### PR DESCRIPTION
**Pull request by habrahamyanpro**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)

 - fixes #2739

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by David-DB88** [comment]
Everything looks great, but when this task should marg it need to be updated testing library to the new version so it's needs to be doing some changes in a New.spec.tsx for example most of act needs to be removed  for example
 await  act(()=>userEvent.type(
      screen.getByLabelText('Maximum message size in bytes *'),
      minValue
    ); 
)} should changed with 
await userEvent.type(
      screen.getByLabelText('Maximum message size in bytes *'),
      minValue
    );

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 1]
```diff
@@ -90,12 +92,42 @@ describe('New', () => {
...
+        retainTime
+      );
+    });
+    await act(() => {
+      userEvent.type(
+        screen.getByPlaceholderText('Maximum message size'),
+        minValue
+      );
+    });
     await act(() => {
```
Comment (David-DB88) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:125]: await userEvent.click(screen.getByText('Create topic'));

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 2]
```diff
@@ -90,12 +92,42 @@ describe('New', () => {
...
       userEvent.click(screen.getByText('Create topic'));
     });
```
Comment (David-DB88) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:127]: You can change this part with 
`await userEvent.click(screen.getByText('Create topic'));`

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 3]
```diff
@@ -90,12 +92,42 @@ describe('New', () => {
...
+        screen.getByPlaceholderText('Replication Factor'),
+        minValue
+      );
+    });
+    await act(() => {
+      userEvent.type(
+        screen.getByPlaceholderText('Time to retain data (in ms)'),
+        retainTime
+      );
+    });
```
Comment (David-DB88) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:118]: `await userEvent.type(
      screen.getByLabelText('Time to retain data (in ms)'),
      retainTime
    );`

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 4]
```diff
@@ -90,12 +92,42 @@ describe('New', () => {
...
+    await act(() => {
+      userEvent.type(
+        screen.getByPlaceholderText('Maximum message size'),
+        minValue
+      );
+    });
```
Comment (David-DB88) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:124]: `
await userEvent.type(
      screen.getByLabelText('Maximum message size in bytes *'),
      minValue
    );
`

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 5]
```diff
@@ -90,12 +92,42 @@ describe('New', () => {
...
+        screen.getByPlaceholderText('Min In Sync Replicas'),
+        minValue
+      );
+    });
+    await act(() => {
+      userEvent.type(
+        screen.getByPlaceholderText('Replication Factor'),
+        minValue
+      );
+    });
```
Comment (David-DB88) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:112]: `await userEvent.type(
      screen.getByLabelText('Replication Factor *'),
      minValue
    );`

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx (hunk 4 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:124]: after you can add this 
`expect(screen.getByText('Create topic')).toBeEnabled();`

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 6]
```diff
@@ -90,12 +92,42 @@ describe('New', () => {
...
+        screen.getByPlaceholderText('Number of partitions'),
+        minValue
+      );
+    });
+    await act(() => {
+      userEvent.type(
+        screen.getByPlaceholderText('Min In Sync Replicas'),
+        minValue
+      );
+    });
```
Comment (David-DB88) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:106]: this need to change like examples below

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 7]
```diff
@@ -90,12 +92,42 @@ describe('New', () => {
     await act(() => {
       userEvent.type(screen.getByPlaceholderText('Topic Name'), topicName);
     });
+    await act(() => {
+      userEvent.type(
+        screen.getByPlaceholderText('Number of partitions'),
+        minValue
+      );
+    });
```
Comment (David-DB88) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:100]: this need to change like examples below

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 8]
```diff
@@ -90,12 +92,42 @@ describe('New', () => {
     await act(() => {
       userEvent.type(screen.getByPlaceholderText('Topic Name'), topicName);
     });
```
Comment (David-DB88) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:94]: this need to change like examples below

**Comment by David-DB88** [comment]
also you can replace this
await act(() => renderComponent(clusterTopicNewPath(clusterName))); 
whit  renderComponent(clusterTopicNewPath(clusterName));

**Comment by David-DB88** [comment]
well done, @Haarolean please can you look?

**Comment by Haarolean** [comment]
> well done

please leave an approving review in this case @David-DB88 :)

**Review by David-DB88 — APPROVED** [review]

**Comment by iliax** [comment]
@David-DB88 I checked deployment 
1. `Min In Sync Replicas`, `Replication Factor`, `Time to retain data (in ms)` and `Maximum message size in bytes` are still required.
As I describe [here](https://github.com/provectus/kafka-ui/issues/2739) they should be optional and should not be passed if not set explicitly.

2. On Edit page clicking on `Add Custom Parameter` causes UI cleared (white page)

**Review by iliax — CHANGES_REQUESTED** [review]
https://github.com/provectus/kafka-ui/pull/2828#issuecomment-1307312438

File: kafka-ui-react-app/src/components/Topics/shared/Form/TopicForm.tsx [hunk 9]
```diff
@@ -79,6 +79,10 @@ const TopicForm: React.FC<Props> = ({
     navigate(clusterTopicPath(clusterName, topicName));
   };
 
+  const getDynamicSource = (key: string) =>
+    config &&
+    config[key as string]?.source === ConfigSource.DYNAMIC_TOPIC_CONFIG;
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/Topics/shared/Form/TopicForm.tsx:84]: why do we have a casting here

Comment (habrahamyanpro) [reply kafka-ui-react-app/src/components/Topics/shared/Form/TopicForm.tsx:84]: Removing `as string` casting, it's not necessary 👌

File: kafka-ui-react-app/src/components/Topics/shared/Form/TopicForm.tsx [hunk 10]
```diff
@@ -147,23 +143,21 @@ const TopicForm: React.FC<Props> = ({
...
-                placeholder="Min In Sync Replicas"
-                min="1"
-                name="minInSyncReplicas"
-              />
-              <FormError>
-                <ErrorMessage errors={errors} name="minInSyncReplicas" />
-              </FormError>
-            </div>
-          )}
+          <div>
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/Topics/shared/Form/TopicForm.tsx:146]: do we need this div tag?

**Review by Mgrdich — APPROVED** [review]

**Pull request merged**